tests/unit/remoteworker/test_remote_worker.py

```diff
@@ -141,3 +148,158 @@ def test_return_url_per_environment(self):
+
+        self.challenge_host_team = {
+            "team_name": "Test Challenge Host Team",
+            "created_by": self.user
+        }
+
+        self.participant_team = {
+            "team_name": "Participant Team for Challenge",
+            "created_by": self.user
+        }
+
+        self.participant = {
+            "user": self.user,
+            "status": Participant.SELF,
+            "team": self.participant_team
+        }
+
+        self.challenge = {
+            "id": self.challenge_pk,
+            "title": "Test Challenge",
+            "creator": self.challenge_host_team,
+        }
+
+        try:
+            os.makedirs("/tmp/evalai")
+        except OSError:
+            pass
+
+        with self.settings(MEDIA_ROOT="/tmp/evalai"):
+            self.challenge_phase = {
+                "id": self.challenge_phase_pk,
+                "name": "Challenge Phase",
+                "description": "Description for Challenge Phase",
+                "leaderboard_public": False,
+                "is_public": True,
+                "challenge": self.challenge,
+                "test_annotation": SimpleUploadedFile(
+                    "test_sample_file.txt",
+                    b"Dummy file content",
+                    content_type="text/plain"
+                )
+            }
+
+        self.submission = {
+            "id": self.submission_pk,
+            "participant_team": self.participant_team,
+            "challenge_phase": self.challenge_phase,
+            "created_by": self.challenge_host_team["created_by"],
+            "status": "submitted",
+            "input_file": SimpleUploadedFile(
+                "user_annotation_file.txt",
+                b"Dummy file content",
+                content_type="text/plain"
+            ),
+            "method_name": "Test Method",
+            "method_description": "Test Description",
+            "project_url": "http://testserver",
+            "publication_url": "http://testserver",
+            "is_public": True
+        }
+
+    def tearDown(self):
+        try:
+            shutil.rmtree("/tmp/evalai")
+        except OSError:
+            pass
+
+    @mock.patch("scripts.workers.remote_submission_worker.SUBMISSION_DATA_DIR", "mocked/dir/submission_{submission_id}")
+    @mock.patch("scripts.workers.remote_submission_worker.os.path.basename", return_value="user_annotation_file.txt")
+    @mock.patch("scripts.workers.remote_submission_worker.get_submission_by_pk")
+    @mock.patch("scripts.workers.remote_submission_worker.get_challenge_by_queue_name")
+    @mock.patch("scripts.workers.remote_submission_worker.get_challenge_phase_by_pk")
+    @mock.patch("scripts.workers.remote_submission_worker.create_dir_as_python_package")
+    @mock.patch("scripts.workers.remote_submission_worker.download_and_extract_file")
+    @mock.patch("scripts.workers.remote_submission_worker.run_submission")
+    def test_process_submission_message_successfully(self, mock_run_submission,
```

Comment:
can we rename this method to `test_process_submission_message` or `test_process_submission_message_success`